shared_model/interfaces/transaction.hpp

```diff
@@ -71,21 +72,57 @@ namespace shared_model {
        * Quorum means how much signatures of account required for performing
        * transaction.
        */
-      virtual const QuorumType &quorum() const;
+      virtual const QuorumType &quorum() const = 0;
 
-      /**
-       * Equality of transactions means equality of hashes only.
-       * This invariant useful for checking transaction in fast way.
-       * @param rhs - other transaction.
-       * @return true if hashes equal.
-       */
-      bool operator==(const Transaction &rhs) const override {
-        return this->hash() == rhs.hash();
+      iroha::model::Transaction *makeOldModel() const {
+        iroha::model::Transaction *oldStyleTransaction =
+            new iroha::model::Transaction();
+        oldStyleTransaction->created_ts = createdTime();
+        oldStyleTransaction->creator_account_id = creatorAccountId();
+        oldStyleTransaction->tx_counter = transactionCounter();
+
+        std::for_each(commands().begin(),
+                      commands().end(),
+                      [oldStyleTransaction](auto &command) {
+                        oldStyleTransaction->commands.emplace_back(
+                            std::shared_ptr<iroha::model::Command>(
+                                command->makeOldModel()));
+                      });
+
+        std::for_each(signatures().begin(),
+                      signatures().end(),
+                      [oldStyleTransaction](auto &sig) {
+                        oldStyleTransaction->signatures.emplace_back(
+                            *sig->makeOldModel());
+                      });
+
+        return oldStyleTransaction;
       }
 
-      iroha::model::Transaction *makeOldModel() const {
-        // TODO implement conversion to old style transaction
-        return nullptr;
+      std::string toString() const {
+        util::PrettyStringBuilder builder;
+        builder.initString("Transaction");
```

Comment:
Also good idea to make chaining for builders. Such as:
`builder.initString(..).appendField(..).appendField` for reduce boilerplate referencing to builder object.